compiler/next/include/chpl/uast/ASTClassesList.h

```diff
@@ -115,17 +115,21 @@ AST_BEGIN_SUBCLASSES(Expression)       // old AST: Expr
     AST_NODE(TupleDecl)
 
     AST_BEGIN_SUBCLASSES(NamedDecl)
-      //AST_NODE(FieldDecl)
       //AST_NODE(ForwardingDecl)
       //AST_NODE(TypeDecl)
 
       AST_NODE(EnumElement)                // old AST: EnumSymbol
-      AST_LEAF(Formal)                     // old AST: ArgSymbol
+
       AST_NODE(Function)                   // old AST: FnSymbol
       AST_NODE(Interface)                  // old AST: InterfaceSymbol
       AST_NODE(Module)                     // old AST: ModuleSymbol
-      AST_LEAF(Variable)                   // old AST: VarSymbol
-                                           // old AST: ShadowVarSymbol
+
+      AST_BEGIN_SUBCLASSES(VarDecl)
```

Comment:
@lydia-duncan @vasslitvinov @e-kayrakli @mppf 

Engin and Lydia both expressed interest in seeing this sort of hierarchy instead of something like

```
Variable
  Field
  Formal
  Local
  Module
  Shadow
```

So this is the ordering I chose.

I ended up using the name `VarDecl` instead of `SingleDecl` out of concerns that:

- Others might confuse `SingleDecl` with the declaration of a variable with the `single` type
- Things like a `Function` could be considered a `SingleDecl` as well, depending on how you look at it

I have concerns that the name `VarDecl` could be confused with `Variable`, but am struggling to think of a better name. Do others have similar concerns, and if so, suggestions or improvements?

Reply:
I'm happy with `VarDecl` but `VarIshDecl` or `VarLikeDecl` would differentiate them from `Variable` if we felt we needed that.

Reply:
This looks good to me. I propose to rename:
* `VarDecl` to `VarLike` (or at least VarLikeDecl)
* `ShadowVariable` to `ShadowVar`